app-typescript/components/ToastMessage.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import ToastText from './ToastText';
+import classNames from 'classnames';
+
+export type MessageProp =
+    | React.ReactNode
+    | string;
+
+export type Position = 'top' | 'bottom' | 'top-right' | 'top-left' | 'bottom-right' | 'bottom-left';
+
+export type NotesType = 'default' | 'primary' | 'success' | 'warning' | 'alert' | 'highlight' | 'light';
+
+export interface IMessageOptions {
+    id?: string;
+    duration?: number | null;
+    type?: NotesType;
+    position?: Position;
+    icon?: string;
+    show?: boolean;
+}
+
+interface IProps extends IMessageOptions {
+    message: MessageProp;
+    closeElement(id, position): void;
+}
+
+interface IState {
+    show: boolean;
+}
+
+ export const ToastMessage = ({
+    id,
+    message,
+    type,
+    icon,
+    duration,
+    position,
+    closeElement,
+}: IProps) => {
+    const [show, setShow] = React.useState(true);
+    const intervalRef = React.useRef();
+    const intervalShow = React.useRef();
+    React.useEffect(() => {
+        TestTimeout(id, duration);
+    });
+
+    function TestTimeout(element: string, timeout: number | null) {
```

Comment:
it seems to me that this should check if show is true before setting up any timer, you call inside `setShow(false)` which will trigger that `useEffect` again and that will setup another timers.

also not sure if you need 2 timers running together, rather than having 1 and let that inner function setup the second one 1s later.